zebra-consensus/src/block.rs

```diff
@@ -123,20 +123,35 @@ where
             // Skip contextual checks for the genesis block
             let previous_block_hash = block.header.previous_block_hash;
             if previous_block_hash != crate::parameters::GENESIS_PREVIOUS_BLOCK_HASH {
-                tracing::debug!(?height, "Awaiting previous block from state");
+                if height == block::Height(0) {
+                    Err("Invalid block: height is 0, but previous block hash is not null.")?;
```

Comment:
This is a style that rustc / clippy use for errors but I think its a good one to adopt here as well, no capitalization or punctuation in the error messages:

```suggestion
                    Err("invalid block: height is 0, but previous block hash is not null")?;
```

Hmm, I wonder if clippy could lint for this...